### PR TITLE
KNL-1346 - Allow space character in urls

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

**Comment by jonespm** [comment]
Comments on inclusion?

**Comment by ottenhoff** [comment]
+1. Thanks for including test.

File: kernel/kernel-impl/src/main/resources/antisamy/high-security-policy.xml [hunk 1]
```diff
@@ -43,8 +43,8 @@
     <regexp name="htmlTitle" value="[\p{L}\p{N}\s\-_',:\[\]!\./\\\(\)&amp;]*" /> <!-- force non-empty with a '+' at the end instead of '*' -->
     <regexp name="htmlClass" value="[a-zA-Z0-9\s,\-_]+" />
 
-    <regexp name="onsiteURL" value="([\p{L}\p{N}\\\.\#@\$%\+&amp;;\-_~,\?=/!]+|\#(\w)+)" />
-    <regexp name="offsiteURL" value="(\s)*((ht|f)tp(s?)://|mailto:)[\p{L}\p{N}]+[\p{L}\p{N}\p{Zs}\.\#@\$%\+&amp;;:\-_~,\?=/!\(\)]*(\s)*" />
+    <regexp name="onsiteURL" value="([\p{L}\p{N}\\\.\#@\$%\s\+&amp;;\-_~,\?=/!]+|\#(\w)+)" />
```
Comment (buckett) [inline kernel/kernel-impl/src/main/resources/antisamy/high-security-policy.xml]: I think `\s` will allow newlines (and others) in URLs, I'm pretty sure this won't be handled well by browsers, is there any reason we need to add more than just a space?

**Comment by jonespm** [comment]
I was going to use just an empty space there there % \+&amp; instead of %\s\+&amp; but I felt that wouldn't be very clear to anyone. Maybe using \p{Blank} would be good, though that includes tab. Would anyone possibly have a tab in a URL? (http://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html)

All of them pass the unit test.

**Comment by buckett** [comment]
I agree `\s` and `\p{Blank}` are both much more readable than ` `, but my concern would be that something like a newline might be interperted by a browser as the end of the tag in a similar way to missing trailing quotes on attributes used to be parsed `<img src="someurl" alt="no quote><h1>Heading</h1>``

**Comment by jonespm** [comment]
Yeah, I'm okay with changing it and just putting the empty space there. I think this is part of the reason they don't want unencoded white space in URL's. ;)

**Comment by jonespm** [comment]
Okay, redid commit with a single space and a comment in there about the space

**Comment by buckett** [comment]
LGTM

**Pull request merged**